test/assocPath.js

```diff
@@ -49,4 +49,9 @@ describe('assocPath', function() {
     eq(assocPath(['foo', 'bar', 'baz'], 42, {foo: null}), {foo: {bar: {baz: 42}}});
   });
 
+  it('supports array indices', function() {
+    eq(assocPath([0], 42, [1, 2, 3]), [42, 2, 3]);
+    eq(assocPath([-1], 42, [1, 2, 3]), [1, 2, 42]);
```

Comment:
For positive indices, sparse makes sense.. but for negative it gets confusing. In Perl it throws and error which seems sane:

    $ perl -e '[]->[-1] = 1'
    Modification of non-creatable array value attempted, subscript -1 at -e line 1.